config/routes.rb

```diff
@@ -1,8 +1,6 @@
-# frozen_string_literal: true
-
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
-end
+    get '/users', to: 'users#index', as: 'users'
+    get '/users/:id', to: 'users#show', as: 'user'
+    get '/users/user_id/posts', to: 'posts#index', as: 'user_posts'
+    get '/users/user_id/posts/:id', to: 'posts#show', as: 'user_post'
```

Comment:
**OPTIONAL** - You have done well so far, but to help make your code even more readable, I suggest you use the [`resource`](https://launchschool.com/books/demystifying_rails/read/routes_and_resources#resourcenotation) notation in Rails. It helps to generate your routes automatically